eureka/S5_lightcurve_fitting/models.py

```diff
@@ -292,28 +304,46 @@ def _parse_coeffs(self, **kwargs):
         """
 
         # Parse 'c#' keyword arguments as coefficients
-        coeffs = np.zeros(10)
+        coeffs = np.zeros((self.nchan,9))
         for k, v in self.parameters.dict.items():
+            remvisnum=k.split('_')
             if k.lower().startswith('c') and k[1:].isdigit():
-                coeffs[int(k[1:])] = v[0]
+                coeffs[0,int(k[1:])] = v[0]
+            elif len(remvisnum)>1:
+                if remvisnum[0].lower().startswith('c') and remvisnum[0][1:].isdigit() and remvisnum[1].isdigit():
+                    coeffs[int(remvisnum[1]),int(remvisnum[0][1:])] = v[0]
 
         # Trim zeros and reverse
-        self.coeffs = np.trim_zeros(coeffs)[::-1]
+        coeffs=coeffs[:,~np.all(coeffs==0,axis=0)]
+        coeffs=np.flip(coeffs,axis=1)
+        self.coeffs=coeffs
+        # self.coeffs = np.trim_zeros(coeffs)[::-1]
 
     def eval(self, **kwargs):
         """Evaluate the function with the given values"""
         # Get the time
         if self.time is None:
             self.time = kwargs.get('time')
 
-        # Create the polynomial from the coeffs
-        poly = np.poly1d(self.coeffs)
+        longparamlist=self.longparamlist
```

Comment:
Why do this local renaming? Why not use `self.nchan` below, and you don't use longparamlist and paramtitles anymore it seems